### ISSUE TITLE
3.15 Ume Script Classification

### ISSUE THREAD

**Issue by ta4tsering** [OpenPecha/BEC#153]
Description:
- Check previous project’s data availability.
- Train models at different levels of the script family tree, starting from broad categories.
- Develop an MVP for script classification tool (Input: Images -> Output: Tags).
- Annotate batches of images provided by BDRC. (7 annotators for 1 month)

Repo: https://github.com/OpenPecha/script-classification-model-train
Note:
- This card can start only after the guidelines from the BDRC
- The guideline will only come in March.

**Illustration**

<img width="2314" height="320" alt="Image" src="https://github.com/user-attachments/assets/53019408-b0c0-48a9-aa5d-f980dae6f0ef" />

**Comment by gangagyatso4364** [comment]
- [ ] exploring general font classification models used for non tibetan english unicoded fonts.

**Comment by gangagyatso4364** [comment]
Current best candidate model image classification is Dinov2, Dinov3 by meta. It can be finetuned on our tibetan script classifcation. 
[medium link ](https://medium.com/digital-mind/visualizing-dinov2-contrastive-learning-and-classification-examples-9e6d8f87acf6)
Experiments can be conducted via Andre Karpathy autoresearch repo. 
[autoresearch repo](https://github.com/karpathy/autoresearch.git)

**Comment by gangagyatso4364** [comment]
| Dataset                                   | Size        | Classes   | Use Case                    |
| ----------------------------------------- | ----------- | --------- | --------------------------- |
| arbml/FontsLarge                          | 9.9K images | 9 fonts   | General font classification |
| arbml/FontsSmall                          | 999 images  | 10 fonts  | Quick testing               |
| gagan3012/hindawi_fonts                   | 79K images  | 19 fonts  | Larger scale                |
| yuanmei424/fonts_en                       | 19.8M       | 5-7 fonts | English fonts               |
| Leeps/Fonts-Individual-Letters            | 63K images  | 36 fonts  | Per-letter classification   |
| biglam/early_printed_books_font_detection | 9.2K        | 33 fonts  | Historical fonts            |

**Comment by gangagyatso4364** [comment]
| Model                    | Params | Finetuneable         | Notes                                    |
| ------------------------ | ------ | -------------------- | ---------------------------------------- |
| ViT (Vision Transformer) | 86M+   | ✅ Yes                | Best overall accuracy, transformer-based |
| ResNet50                 | 25M    | ✅ Yes                | Classic, reliable, good baseline         |
| EfficientNet-B0          | 5.3M   | ✅ Yes                | Great accuracy/size tradeoff             |
| ConvNeXt                 | 88M    | ✅ Yes                | Modern CNN, competitive with ViT         |
| MobileNetV3              | 5.5M   | ✅ Yes                | Lightweight, for edge devices            |
| CLIP                     | 63M    | ✅ Yes (linear probe) | Zero-shot potential                      |

**Comment by webuddhist-bot1** [comment]
## 📋 Card Review

**Reviewed by:** Webuddhist OpenClaw Bot

### Summary
- Card Type: Developer
- Status: Needs Improvement

### Issues Found
- ❗ Missing Completion Criteria or Definition of Done
- ⚠️ No Implementation Plan
- ⚠️ No Subtasks

### Suggestions
- Add Completion Criteria (e.g., "MVP produces accurate script tags for X% of test images")
- Break into subtasks (data check, train model, build MVP, annotation workflow)
- Add a User Story

### Good Points
- Title is clear and specific
- Description covers major work areas
- Note correctly identifies dependency on BDRC guidelines

---
*This review was automatically generated by Webuddhist OpenClaw Bot*

**Comment by webuddhist-bot1** [comment]
### Summary
- Card Type: Data Team
- Status: Ready

### Good Points
- Description is clear and detailed with4 meaningful bullet points
- Implementation plan is well-structured with sequential steps
- Subtasks cover both model training and human annotation aspects
- Includes note about dependencies (BDRC guidelines) - good communication
- Timeline mentioned for annotation phase (7 annotators, 1 month)

### Issues Found
- ℹ️ User Story format is missing (preferred for Data Team cards)
- ℹ️ Definition of Done is not explicitly stated

### Suggestions
- Consider adding a user story in format: "As a [user], I want [goal], so that [benefit]"
- Define what "success" looks like for the MVP (e.g., accuracy threshold, number of scripts classified)

The card is actionable and ready to start once BDRC guidelines are available.

**Comment by webuddhist-bot1** [comment]
### Card Review Summary
- **Card Type:** Data Team
- **Status:** Needs Improvement

### Issues Found
- ⚠️ Missing Definition of Done - No clear, measurable completion criteria defined
- ⚠️ Missing Data Source link - References BDRC images but no link to previous project or existing dataset
- ⚠️ Blocked task - Card notes depend on BDRC guidelines arriving in March (now April); clarify current status

### Suggestions
- Add explicit completion criteria (e.g., "Model achieves X% accuracy on test set")
- Link to previous project's data availability results
- Confirm whether BDRC guidelines have been received
- Specify expected output format for the MVP

### Good Points
- Clear description of scope (4 subtasks enumerated)
- Title is specific and describes the work
- Assignee is named
